pkg/eventgenerator/http/generator.go

```diff
@@ -0,0 +1,92 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Generator struct {
+	clientset *kubernetes.Clientset
+	config    *rest.Config
+	logger    logger.Logger
+	namespace string
+	podName   string
+}
+
+func NewGenerator(config *rest.Config, log logger.Logger) (*Generator, error) {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+	return &Generator{
+		clientset: clientset,
+		config:    config,
+		logger:    log,
+		namespace: "default",
+		podName:   fmt.Sprintf("http-eventgen-pod-%d", time.Now().Unix()),
+	}, nil
+}
+
+func (h *Generator) Generate(params map[string]string, count, interval string) (error) {
+	pods, err := h.clientset.CoreV1().Pods(h.namespace).List(context.Background(), metav1.ListOptions{})
+    if err != nil {
+        return fmt.Errorf("failed to list pods: %v", err)
+    }
+
+    // Check for any pod with the dns-eventgen-pod prefix
+    for _, pod := range pods.Items {
+        if strings.HasPrefix(pod.Name, "http-eventgen-pod-") {
+            return fmt.Errorf("HTTP event generator is already running with pod %s, please stop it first", pod.Name)
+        }
+    }
+	
+	url, ok := params["url"]
+	if !ok {
+		return fmt.Errorf("url parameter is required for HTTP event generator")
+	}
+
+	command := fmt.Sprintf("i=1; while [ $i -le %s ] || [ %s -eq -1 ]; do curl -s -o /dev/null -w '%%{http_code}\\n' -L %s; i=$((i+1)); sleep %s; done", count, count, url, interval)
+
+	container := corev1.Container{
+		Name:    "http-eventgen-container",
+		Image:   "alpine/curl",
+		Command: []string{"sh", "-c", command},
+	}
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: h.podName},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers:    []corev1.Container{container},
+		},
+	}
+
+	_, err = h.clientset.CoreV1().Pods(h.namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create pod: %v", err)
+	}
+
+	h.logger.Debugf("Created HTTP event generation pod: %s", h.podName)
+	return nil
+}
+
+func (h *Generator) Cleanup() (string, error) {
+	if h.podName == "" {
+		return "", fmt.Errorf("pod %s is not found in %s namespace", h.podName, h.namespace)
```

Comment:
This error mesage is confusing for me.

When the pod name is empty, why is an error message displayed that the empty pod name is not found in the namespace

Reply:
Yes actually I put it for the case when user deleted that namespace/pod on his own by mistake or intentionally. Should I remove it?